munit/shared/src/main/scala/munit/Assertions.scala

```diff
@@ -90,8 +90,20 @@ trait Assertions {
   }
 
   def intercept[T <: Throwable](
-      body: => Any,
-      msg: String = ""
+      body: => Any
+  )(implicit T: ClassTag[T], loc: Location): T = {
+    _intercept(None,body)
+  }
+
+  def interceptMessage[T <: Throwable](msg: String)(
+      body: => Any
+  )(implicit T: ClassTag[T], loc: Location): T = {
+    _intercept(Some(msg),body)
+  }
+
+  private def _intercept[T <: Throwable](
```

Comment:
It's not idiomatic to use `_name` in Scala
```suggestion
  private def runIntercept[T <: Throwable](
         body: => Any,
         expectedExceptionMessage: Option[String]
```

Reply:
done